include/mega/utils.h

```diff
@@ -408,6 +408,59 @@ struct CacheableReader
     void eraseused(string& d); // must be the same string, unchanged
 };
 
+template<typename T>
+void hashCombine(size_t& seed, const T& v)
+{
+    // Taken from Boost's hash combine function
+    seed ^= std::hash<T>{}(v) + 0x9e3779b9 + (seed<<6) + (seed>>2);
+}
+
+/////// Following are a few helpers that are required for compile-time `forEach` further down
+
+template<std::size_t...>
+struct Indices
+{};
+
+template<std::size_t...>
+struct ConstructRange;
+
+template<std::size_t end, std::size_t idx, std::size_t... i>
+struct ConstructRange<end, idx, i...> : ConstructRange<end, idx + 1, i..., idx>
+{};
+
+template<std::size_t end, std::size_t... i>
+struct ConstructRange<end, end, i...>
+{
+    using type = Indices<i...>;
+};
+
+template<std::size_t b, std::size_t e>
+struct IndexRange
+{
+    using type = typename ConstructRange<e, b>::type;
+};
+
+template<typename Container, typename Functor>
+void forEachIndex(Indices<>, Container&&, Functor&&)
+{}
+
+template<std::size_t i, std::size_t... j, typename Container, typename Functor>
+void forEachIndex(Indices<i, j...>, Container&& container, Functor&& functor)
+{
+    std::forward<Functor>(functor)(std::get<i>(std::forward<Container>(container)));
+    forEachIndex(Indices<j...>{}, std::forward<Container>(container), std::forward<Functor>(functor));
+}
+
+/////// forEach over a std::tuple, unrolled at compile time
+
+template<typename Tuple, typename Functor>
+void forEach(Tuple&& tup, Functor&& functor)
+{
+    constexpr auto size = std::tuple_size<typename std::decay<Tuple>::type>::value;
+    using IndexType = typename IndexRange<0, size>::type;
+    forEachIndex(IndexType{}, std::forward<Tuple>(tup), std::forward<Functor>(functor));
+}
+
```

Comment:
As this is only used from the tests, can we put it in the test headers rather than as part of the main SDK please?

Reply:
This should be part of production code and I am soon planning on using it to iterate over a tuple.

Reply:
I just added another change that uses `forEach` in `MegaApiImpl::getregisteredcontacts_result`. I ran the tests hitting this code and they're still passing. 

Sorry for this not being related to the actual fix anymore but I thought it would still be useful to have.